
#Estimate mode
use_gnss_mode: RTKLIB
## RTKLIB (rtklib) or NMEA (nmea)
## The NMEA(nmea) mode eliminates the need to enter topics of type rtklib into eagleye, but may reduce accuracy.

use_canless_mode: false  #Estimate mode

twist_topic: /can_twist
imu_topic: /imu/data_raw
rtklib_nav_topic: /rtklib_nav
nmea_sentence_topic: /nmea_sentence
gga_topic: /navsat/gga
rmc_topic: /navsat/rmc
localization_pose_topic: /localization/pose # This parameter is used to input external poses such as map matching to eagleye. (Normally it is not used.)

tf_gnss_frame:
  parent: "base_link"
  child: "gnss"

ecef_base_pos:                                        #The position in the ecef coordinate system to use when converting to the enu coordinate system. If 0, the position entered for the first time when starting the program will be used as a reference.
  x : 0.0                               
  y : 0.0
  z : 0.0
  use_ecef_base_position : false

common:
  imu_rate: 50
  gnss_rate: 5
  stop_judgment_threshold: 0.01
  slow_judgment_threshold: 0.278
  moving_judgment_threshold: 2.78
  
velocity_scale_factor:                                #Parameters for estimating vehicle speed scale factor.
  estimated_minimum_interval: 20
  estimated_maximum_interval: 400
  gnss_receiving_threshold: 0.25
  velocity_scale_factor_save_str: /config/velocity_scale_factor.txt
  save_velocity_scale_factor: false                     #If this parameter is set to true, the velocitry_scale_factor saved at the last startup is used as the initial value at startup.
  velocity_scale_factor_save_duration: 100.0           #[sec]
  th_velocity_scale_factor_percent: 10.0              #If the estimated value is too large or too small than this percentage, it is considered abnormal.

yawrate_offset_stop:                                  #Parameters related to estimation of yaw rate offset due to temperature drift and noise of IMU during stoppage
  estimated_interval: 4
  outlier_threshold: 0.002

angular_velocity_offset_stop:                         #Parameters related to estimation of yaw rate offset due to temperature drift and noise of IMU during stoppage
  estimated_interval: 4
  outlier_threshold: 0.002

yawrate_offset:                                       #Parameters related to estimation of yaw rate offset due to temperature drift and noise of IMU while driving
  estimated_minimum_interval: 30
  gnss_receiving_threshold: 0.25
  outlier_threshold: 0.002
  1st:
    estimated_maximum_interval: 300
  2nd:
    estimated_maximum_interval: 500

heading:                                              #Parameters for estimating the azimuth of a car
  estimated_minimum_interval: 10
  estimated_maximum_interval: 30
  gnss_receiving_threshold: 0.25
  outlier_threshold: 0.0524                           #Threshold for ending estimated outlier rejection. (default:0.0524 rad = 3 degree)
  outlier_ratio_threshold: 0.5
  curve_judgment_threshold: 0.0873

rtk_heading:                                          #Parameters for estimating the azimuth of a car
  update_distance: 0.3
  estimated_minimum_interval: 10
  estimated_maximum_interval: 30
  gnss_receiving_threshold: 0.25
  outlier_threshold: 0.0524
  outlier_ratio_threshold: 0.5
  curve_judgment_threshold: 0.0873

heading_interpolate:
  sync_search_period: 2

slip_angle:
  manual_coefficient: 0.0                             #If you do not want to correct slip angle, set slip angle estimate to false and set the coefficient here to 0

slip_coefficient:
  estimated_minimum_interval: 2
  estimated_maximum_interval: 100
  curve_judgment_threshold: 0.017453
  lever_arm: 0.26                                     #Distance from center of rear wheel axle to GNSS antenna. (m)

trajectory:
  deadlock_threshold: 1
  timer_updata_rate: 10                               #default:10 hz
  curve_judgment_threshold: 0.017453

position:
  estimated_interval: 300
  update_distance: 0.1
  outlier_threshold: 3.0
  gnss_receiving_threshold: 0.25
  outlier_ratio_threshold: 0.5

position_interpolate:
  sync_search_period: 2

smoothing:
  moving_average_time: 3
  moving_ratio_threshold: 0.1

height:
  estimated_minimum_interval: 200
  estimated_maximum_interval: 2000
  update_distance: 0.1
  gnss_receiving_threshold: 0.1
  outlier_threshold: 0.3
  outlier_ratio_threshold: 0.5
  moving_average_time: 1

enable_additional_rolling:
  update_distance: 0.3
  moving_average_time: 1
  sync_judgment_threshold: 0.01
  sync_search_period: 1

rolling:
  filter_process_noise: 0.01
  filter_observation_noise: 1

velocity_estimator:
  gga_downsample_time: 1
  stop_judgment_velocity_threshold: 0.2 
  stop_judgment_interval: 1
  variance_threshold: 0.000025
  pitchrate_offset:
    estimated_interval: 8
  pitching:
    estimated_interval: 5
    outlier_threshold: 0.0174
    gnss_receiving_threshold: 0.2
    outlier_ratio_threshold: 0.5
  acceleration_offset:
    estimated_minimum_interval: 30
    estimated_maximum_interval: 500
    filter_process_noise: 0.01
    filter_observation_noise: 1
  doppler_fusion:
    estimated_interval: 4
    gnss_receiving_threshold: 0.2
    outlier_ratio_threshold: 0.5
    outlier_threshold: 0.1

monitor:
  print_status: true
  log_output_status: false
  update_rate: 10.0
  th_gnss_deadrock_time: 10.0
  use_compare_yawrate: false                        #Whether to compare the yaw rate with the yaw rate calculated from the vehicle speed model
  comparison_twist_topic: /calculated_twist
  th_diff_rad_per_sec: 0.17453                      #Threshold of difference from comparison twist yaw rate(0.17453 rad/s = 10 degree/s)
  th_num_continuous_abnormal_yawrate: 25            #Threshold for consecutive abnormal values in comparison twist
  th_dr_distance: 50.0                              #In rtk dead reckoning mode, if the distance traveled for this parameter in RTK no fix,
                                                    #an emergency will be output. [m]
